tests/schedule.go

```diff
@@ -348,24 +360,111 @@ func (s *ScheduleFunctionalSuite) TestBasics() {
 	s.NoError(err)
 
 	// wait for one new run
-	s.Eventually(func() bool { return atomic.LoadInt32(&runs2) == 1 }, 7*time.Second, 500*time.Millisecond)
+	s.Eventually(
+		func() bool { return atomic.LoadInt32(&runs2) == 1 },
+		7*time.Second,
+		500*time.Millisecond,
+	)
 
 	// describe again
-	describeResp, err = s.engine.DescribeSchedule(NewContext(), &workflowservice.DescribeScheduleRequest{
-		Namespace:  s.namespace,
-		ScheduleId: sid,
-	})
+	describeResp, err = s.engine.DescribeSchedule(
+		NewContext(),
+		&workflowservice.DescribeScheduleRequest{
+			Namespace:  s.namespace,
+			ScheduleId: sid,
+		},
+	)
 	s.NoError(err)
 
-	s.Equal(schSAValue.Data, describeResp.SearchAttributes.IndexedFields[csa].Data)
+	s.Len(describeResp.SearchAttributes.GetIndexedFields(), 3)
+	s.Equal(schSAValue.Data, describeResp.SearchAttributes.IndexedFields[csaKeyword].Data)
+	s.Equal(schSAIntValue.Data, describeResp.SearchAttributes.IndexedFields[csaInt].Data)
+	s.Equal(schSABoolValue.Data, describeResp.SearchAttributes.IndexedFields[csaBool].Data)
 	s.Equal(schMemo.Data, describeResp.Memo.Fields["schedmemo1"].Data)
-	s.Equal(wfSAValue.Data, describeResp.Schedule.Action.GetStartWorkflow().SearchAttributes.IndexedFields[csa].Data)
+	s.Equal(wfSAValue.Data, describeResp.Schedule.Action.GetStartWorkflow().SearchAttributes.IndexedFields[csaKeyword].Data)
 	s.Equal(wfMemo.Data, describeResp.Schedule.Action.GetStartWorkflow().Memo.Fields["wfmemo1"].Data)
 
 	s.DurationNear(describeResp.Info.UpdateTime.AsTime().Sub(updateTime), 0, 3*time.Second)
 	lastAction := describeResp.Info.RecentActions[len(describeResp.Info.RecentActions)-1]
 	s.True(lastAction.ScheduleTime.AsTime().UnixNano()%int64(5*time.Second) == 1000000000, lastAction.ScheduleTime.AsTime().UnixNano())
 
+	// update schedule and search attributes
+
+	schedule.Spec.Interval[0].Phase = durationpb.New(1 * time.Second)
+	schedule.Action.GetStartWorkflow().WorkflowType.Name = wt2
+
+	schSAIntValue, _ = payload.Encode(321)
+	_, err = s.engine.UpdateSchedule(NewContext(), &workflowservice.UpdateScheduleRequest{
+		Namespace:  s.namespace,
+		ScheduleId: sid,
+		Schedule:   schedule,
+		Identity:   "test",
+		RequestId:  uuid.New(),
+		SearchAttributes: &commonpb.SearchAttributes{
+			IndexedFields: map[string]*commonpb.Payload{
+				csaKeyword: schSAValue,
+				csaInt:     schSAIntValue,
+			},
+		},
+	})
+	s.NoError(err)
+
+	// wait for one new run
```

Comment:
do we have to wait for one run? can we just do Eventually on DescribeSchedule and wait for the search attributes?